great_expectations/execution_engine/execution_engine.py

```diff
@@ -527,6 +458,208 @@ def resolve_data_reference(
+                    )
+                )
+            else:
+                metric_fn_type: MetricFunctionTypes = getattr(
+                    metric_fn, "metric_fn_type", MetricFunctionTypes.VALUE
+                )
+                if isinstance(
+                    metric_fn_type, MetricFunctionTypes
+                ) and metric_fn_type not in [
+                    MetricPartialFunctionTypes.MAP_FN,
+                    MetricPartialFunctionTypes.MAP_SERIES,
+                    MetricPartialFunctionTypes.MAP_CONDITION_FN,
+                    MetricPartialFunctionTypes.MAP_CONDITION_SERIES,
+                    MetricPartialFunctionTypes.WINDOW_FN,
+                    MetricPartialFunctionTypes.WINDOW_CONDITION_FN,
+                    MetricPartialFunctionTypes.AGGREGATE_FN,
+                    MetricFunctionTypes.VALUE,
+                ]:
+                    logger.warning(
+                        f'Unrecognized metric function type while trying to resolve "{metric_to_resolve.id}".'
+                    )
+                metric_fn_direct_configurations.append(
+                    MetricComputationConfiguration(
+                        metric_configuration=metric_to_resolve,
+                        metric_fn=metric_fn,
+                        metric_provider_kwargs=metric_provider_kwargs,
+                    )
+                )
+
+        return metric_fn_direct_configurations, metric_fn_bundle_configurations
+
+    def _get_computed_metric_evaluation_dependencies_by_metric_name(
+        self,
+        metric_to_resolve: MetricConfiguration,
+        metrics: Dict[Tuple[str, str, str], MetricValue],
+    ) -> Dict[str, Union[MetricValue, Tuple[Any, dict, dict]]]:
+        """
+        Gathers resolved (already computed) evaluation dependencies of metric-to-resolve (not yet computed)
+        "MetricConfiguration" object by "metric_name" property of resolved "MetricConfiguration" objects.
+
+        Args:
+            metric_to_resolve: dependent (not yet resolved) "MetricConfiguration" object
+            metrics: resolved (already computed) "MetricConfiguration" objects keyd by ID of that object
+
+        Returns:
+            Dictionary keyed by "metric_name" with values as computed metric or partial bundling information tuple
+        """
+        metric_dependencies_by_metric_name: Dict[
+            str, Union[MetricValue, Tuple[Any, dict, dict]]
+        ] = {}
+
+        metric_name: str
+        metric_configuration: MetricConfiguration
+        for (
+            metric_name,
+            metric_configuration,
+        ) in metric_to_resolve.metric_dependencies.items():
+            if metric_configuration.id in metrics:
+                metric_dependencies_by_metric_name[metric_name] = metrics[
+                    metric_configuration.id
+                ]
+            elif self._caching and metric_configuration.id in self._metric_cache:  # type: ignore[operator] # TODO: update NoOpDict
```

Comment:
What exactly is the error message here?

Reply:
@NathanFarmer It was there prior.  I am actually not sure.